### PR TITLE
Fix for issue #42

### PR DESCRIPTION
**Pull request by MarkSonghurst**
Proposed fix for issue #42 
I've run results_test.go successfully, but I don't have access to a pubs database to run the complete set of tests. 
Please can someone do this for me.
Thanks.

### CODE REVIEW

File: fetch.go [hunk 1]
```diff
@@ -68,11 +68,14 @@ func (conn *Conn) fetchResults() ([]*Result, error) {
 		}
 
 		for i := 0; ; i++ {
-			rowCode := C.dbnextrow(conn.dbproc)
-			if rowCode == C.NO_MORE_ROWS {
+			switch rowCode := C.dbnextrow(conn.dbproc); rowCode {
+			case C.NO_MORE_ROWS:
 				break
```
Comment (ianic) [inline fetch.go]: This break terminates execution of the switch statement. 
That changes the behavior.
C.NO_MORE_ROWS status should terminate outer for statement.

**Comment by ianic** [comment]
Test are not passing.
Problem is in the changed behavior of the for loop.

**Comment by MarkSonghurst** [comment]
OK, thanks for the review. 
Now #44 has been merged I will set up my own PUBS on Azure and address the issues.

**Comment by MarkSonghurst** [comment]
@ianic I'm using SQL Server for Linux under Docker, so I can now run the tests against pubs.
I've corrected the infinite loop that was breaking the tests, they all pass on my server.
Sorry for all the commits!

**Comment by ianic** [comment]
It is great to hear that you, as a biggest contributor to the project, now has testing environment.

**Pull request merged**